src/signals/my-incidents/hooks/usePostEmail.ts

```diff
@@ -1,23 +1,25 @@
 // SPDX-License-Identifier: MPL-2.0
 // Copyright (C) 2022 Gemeente Amsterdam
-import { useCallback } from 'react'
+import { useCallback, useEffect, useState } from 'react'
 
 import useFetch from 'hooks/useFetch'
-import type { FetchResponse } from 'hooks/useFetch'
 import configuration from 'shared/services/configuration/configuration'
 
+import { isFetchError, isResponseError } from '../types'
+
 type PostEmail = (email: string) => Promise<void>
 
 export const usePostEmail = (): [
   PostEmail,
-  Omit<FetchResponse<null>, 'post'>
+  { errorMessage: string | null }
 ] => {
-  const { post, ...rest } = useFetch<null>()
+  const { post, error } = useFetch<null>()
+  const [errorMessage, setErrorMessage] = useState<string | null>(null)
```

Comment:
might be better to initialise as empty string because its a primitive value and its a lot cleaner